pyquil/quil.py

```diff
@@ -741,16 +742,6 @@ def __add__(self, other: InstructionDesignator) -> "Program":
         p.inst(other)
         return p
 
-    def __iadd__(self, other) -> "Program":
-        """
-        Concatenate two programs together by appending the right-hand side to the left.
-
-        :param other: Another program or instruction to concatenate to this one.
-        :return: A newly concatenated program.
-        """
-        p = Program(other)
-        return self._program.add_instructions(p._program.instructions)
```

Comment:
I removed this because it wasn't exactly the implementation we needed, and there was no clear optimization over the default `__add__` behavior with the correct implementation.